v-next/hardhat/test/internal/builtin-plugins/network-manager/network-manager.ts

```diff
@@ -355,431 +363,1943 @@ describe("NetworkManagerImplementation", () => {
+                }),
+              );
+
+              assertValidationErrors(validationErrors, [
+                {
+                  path: ["networks", "hardhat", "accounts", 0],
+                  message: `Expected a hex-encoded private key or a Configuration Variable`,
+                },
+                {
+                  path: ["networks", "hardhat", "accounts", 1],
+                  message: `Expected a hex-encoded private key or a Configuration Variable`,
+                },
+              ]);
+            });
+
+            it("should not allow invalid private keys", async () => {
+              const validationErrors = await validateNetworkUserConfig(
+                httpConfig({
+                  accounts: [
+                    "0xgggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggg",
+                  ],
+                }),
+              );
+
+              assertValidationErrors(validationErrors, [
+                {
+                  path: ["networks", "hardhat", "accounts", 0],
+                  message:
+                    "Expected a hex-encoded private key or a Configuration Variable",
+                },
+              ]);
+            });
+          });
         });
 
-        it("should allow an account with a valid HttpNetworkHDAccountsConfig", async () => {
-          networkConfig.accounts = {
-            mnemonic: "asd asd asd",
-            initialIndex: 0,
-            count: 123,
-            path: "m/123",
-            passphrase: "passphrase",
-          };
+        it("should fail with invalid types", async () => {
+          const accounts = [123, [{}], { asd: 123 }];
 
-          const validationErrors =
-            await validateNetworkUserConfig(networkConfig);
+          const validationErrors = await validateNetworkUserConfig(
+            httpConfig({ accounts }),
+          );
 
-          assert.equal(validationErrors.length, 0);
+          assertValidationErrors(validationErrors, [
+            {
+              path: ["networks", "hardhat", "accounts", 0],
+              message:
+                "Expected a hex-encoded private key or a Configuration Variable",
+            },
+            {
+              path: ["networks", "hardhat", "accounts", 1],
+              message:
+                "Expected a hex-encoded private key or a Configuration Variable",
+            },
+            {
+              path: ["networks", "hardhat", "accounts", 2],
+              message:
+                "Expected a hex-encoded private key or a Configuration Variable",
+            },
+          ]);
         });
 
-        it("should allow valid private keys with missing hex prefix", async () => {
-          networkConfig.accounts = [
-            "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
+        it("should fail with invalid HttpNetworkHDAccountsConfig", async () => {
+          const accountsValuesToTest: Array<
+            [Partial<any>, ExpectedValidationError[]]
```

Comment:
isn't `Partial<any>` pretty much the same as `any`?

Reply:
Yes, this appears to be a mistake. I'll push an update!